File: app/src/main/java/com/dev/cameronc/hues/Home/HomeContract.java
package com.dev.cameronc.hues.Home;

/**
 * Created by ccord on 11/9/2016.
 */

public interface HomeContract
{
    interface View
    {
        void showBridges(String bridges);
    }

    interface Presenter<V> extends com.dev.cameronc.hues.Base.Presenter<View>
    {

    }
}

File: app/src/main/java/com/dev/cameronc/hues/Home/HomeContract.kt
package com.dev.cameronc.hues.Home

import com.philips.lighting.model.PHGroup

/**
 * Created by ccord on 11/9/2016.
 */

interface HomeContract
{
    interface View
    {
        fun navigateToConnectScreen()
        fun notifyBridgeConnected()
        fun showLightGroups(allGroups: List<LightGroupAdapter.HueGroupInfo>)
        fun showNoLightGroups()
        fun navigateToGroupScreen(hueGroup: PHGroup)
    }

    interface Presenter : com.dev.cameronc.hues.Base.Presenter<View>
    {
        fun onPresenterDestroyed()
        fun onSliderChanged(event: GroupUpdateEvent)
        fun onGroupOnToggled(phGroup: PHGroup, on: Boolean)
        fun onGroupClicked(hueGroup: PHGroup)
    }
}
